src/models/GOKU.jl

```diff
@@ -5,8 +5,21 @@
 
 struct GOKU <: LatentDE end
 
+@doc raw"""
+    apply_feature_extractor(encoder, x)
+Converts a batch of the initial high-dimensional data into lower-dimensional data (i.e. extracts features)
+# Arguments
+encoder: Has type Encoder{GOKU} or Encoder{LatentODE}\
+x: Initial data, has size (seq_len,), x[1] has size (input_dim, batch_size)
```

Comment:
I would say input data instead of initial data.

I would avoid saying the specific variable names, such as `seq_len`, `input_dim` and `batch_size` because these are the names that we gave them in our training script but the user may chose another name. Although they are pretty clear what they refer to. I'm not sure what we should do here.

Also, the way it is now, the x has no specific type constraint in the function arguments. Maybe we should declare its type Vector{Array}. But I'm not sure what is better in terms of Julia design patterns. I'll think about this. If we add this type declaration we can describe x as: each element of the Vector corresponds to a different time point, and each element consists of an Array of size (input_dim, batch_size), containing the data.

However, saying that the size is (input_dim, batch_size), limits future new methods which includes convolutional networks. But maybe it is ok of now.

Reply:
Actually, we could delcare the types of the apply_feature_extractor as following:

apply_feature_extractor(encoder::Encoder{GOKU}, x::Vector{Matrix}) = encoder.feature_extractor.(x)

(and the same for LatentODE)

This way we will be able in the future to define other methods for convolutional feature extractors, like

apply_feature_extractor(encoder::Encoder{GOKU}, x::Vector{Array) = .....

Reply:
Okay, added type for x, see above comment about type for encoder.  As for the description, how about "x: Input data, has size (<time series length>,), x[1] has size (<input data dimension>, <batch size>)"?

Reply:
Testing <hello>.

Reply:
Testing \<hello\>.

Reply:
I meant, As for the description, how about "x: Input data, has size (\<time series length\>,), x[1] has size (\<input data dimension\>, \<batch size\>)"?

Reply:
Okay, added type for x, see above comment about type for encoder.  As for the description, how about "x: Input data, has size (\<time series length\>,), x[1] has size (\<input data dimension\>, \<batch size\>)"?

Reply:
Good improvemente! How about:

"x: Input data, each of its compounding matrices corresponds to different time frames, and have size `input data dimension` x  `batch size`.

Reply:
How about:

x: Input data, consisting of matrices corresponding to the time frames and having size (`input data dimension` x `batch size`)